### ISSUE TITLE
perf(training): add optional mixed-precision (AMP) path to predictive planner custom training loop

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6509]
Parent issue: #6482 (finding #2). Sibling #6488 owns finding #1 (predictive-planner DataLoader); the replay-buffer finding #3 is a separate child.

Add an opt-in, CUDA-gated automatic mixed-precision path to the predictive-planner custom training loop to demonstrate mixed-precision training, satisfying the parent acceptance criterion 'At least one custom training loop demonstrates AMP usage'. Defaults to off so CPU and existing deterministic paths are unchanged.

```yaml
archetype: implementation
evidence_tier: smoke
linked_policy:
  - docs/dev_guide.md
```

## Scope
- In scripts/training/train_predictive_planner.py, add an opt-in AMP flag (config and/or CLI) defaulting to off.
- When enabled and device is CUDA, wrap the forward/loss computation in _run_epoch (forward/backward region ~lines 372-387) in torch.autocast(device_type='cuda', dtype=torch.float16) and run backward/optimizer.step through torch.cuda.amp.GradScaler.
- When AMP is off or device is CPU, behavior must be identical to today.
- Record the effective AMP setting in the training manifest alongside the existing 'device' field (~line 1064).

## Non-goals
- No replay-buffer change (separate child).
- No DataLoader/pin_memory/num_workers change (owned by #6488).
- No model architecture, optimizer, loss, dataset, or training-objective change.
- No GPU throughput/quality claim; any measured GPU result is a separate follow-up, not merge evidence.

## Definition of done
- [ ] AMP is opt-in, defaults off, and is CUDA-gated.
- [ ] CPU/dry-run training is bit-for-bit unchanged when AMP is off.
- [ ] Effective AMP setting recorded in the training manifest.
- [ ] Focused tests added; ruff check/format pass.

## Factory v2 implementation contract
```json
{
  "allowed_paths": [
    "scripts/training/train_predictive_planner.py",
    "tests/training"
  ],
  "forbidden_paths": [
    "scripts/training/train_distributional_rl.py",
    "robot_sf/training/diffusion_policy.py",
    "robot_sf/training/oracle_imitation_bc_smoke.py",
    "robot_sf/benchmark",
    "scripts/benchmark",
    "configs/benchmark"
  ],
  "objective": "In scripts/training/train_predictive_planner.py, add an opt-in AMP flag (config and/or CLI, default off). When enabled and the device is CUDA, wrap the forward/loss computation in _run_epoch (forward/backward region ~lines 372-387) in torch.autocast(device_type='cuda', dtype=torch.float16) and run backward/optimizer.step through torch.cuda.amp.GradScaler; when AMP is off or the device is CPU, behavior must be identical to today. Record the effective AMP setting in the training manifest alongside the existing 'device' field (~line 1064).",
  "stop_conditions": [
    "AMP must be opt-in and CUDA-gated: never active on CPU or when torch.cuda.is_available() is False.",
    "No change to model architecture, optimizer, loss function, DataLoader (owned by #6488), dataset, horizon weights, or training objective; GradScaler must be a strict no-op when AMP is off.",
    "CPU dry-run/training remains deterministic and bit-for-bit unchanged when AMP is off; no GPU hardware is required to validate.",
    "GPU throughput or quality claims are out of scope for this smoke-tier child; record any GPU measurement as a separate follow-up, not as merge evidence.",
    "Edit only the forward/backward region of _run_epoch and the manifest/config surfaces, not the DataLoader region (lines 304-340) that #6488 owns; if #6488 is unmerged, sequence after it or rebase to avoid merge conflicts on train_predictive_planner.py. This child is file-independent of the replay-buffer child (which edits train_distributional_rl.py) and may run as a parallel lane with it."
  ],
  "validation": [
    "Add or extend a focused test under tests/training asserting AMP defaults to off and that the CPU training path (loss values and step count) is unchanged versus current behavior when AMP is disabled.",
    "Add a test asserting GradScaler/autocast are only constructed/active when AMP is enabled AND torch.cuda.is_available() is True, mocking CUDA availability so the test runs on CPU-only hosts.",
    "uv run pytest tests/training -k predictive -v",
    "uv run ruff check scripts/training/train_predictive_planner.py tests/training",
    "uv run ruff format --check scripts/training/train_predictive_planner.py tests/training",
    "git diff --check"
  ]
}
```
